### PR TITLE
move hot plate item placement stuff into its own system

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
and made xenoarch, hotplate use it
can be used for plates in the future

only visible change is that traversal distorter is now a placeable surface
artifact analyzer is not since placing items takes priority over linking for some reason

**Media**

xenoarch stuff:

https://github.com/space-wizards/space-station-14/assets/39013340/6e22bdef-8df4-4ee1-bd75-c5b00d89e7de

hotplate work:

https://github.com/space-wizards/space-station-14/assets/39013340/9d83db43-541b-455a-a516-3796625473f9

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
no cl no fun

### CODE REVIEW

**Comment by Partmedia** [comment]
- Please justify the YAML-breaking change by giving examples of future systems that could conceivably use the item placement system.
- Or, allow the old solution heater system datafields to still work by making the solution heater component inherit from the new item placement system.
- Please include a video demonstrating the tests done to make sure that the hot plate still works.

**Review by keronshb — CHANGES_REQUESTED** [review]
I agree please make a video.

Also I also feel like while yes generalizing something is good because it makes it less tightly coupled, there should be a reason for this change. A reason being another mechanic that's going to use it added with this PR.

**Comment by 0x6273** [comment]
Could this work for the artifact scanner also?

> Or, allow the old solution heater component datafields to still work by making the solution heater component inherit from the new item placement component.

That's dumb, don't do this.

File: Content.Shared/Placeable/ItemPlacerComponent.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+﻿using Content.Shared.Whitelist;
+
+namespace Content.Shared.Placeable;
+
+/// <summary>
+/// Detects items placed on it that match a whitelist.
+/// </summary>
+[RegisterComponent]
```
Comment (metalgearsloth) [inline Content.Shared/Placeable/ItemPlacerComponent.cs:8]: networked too

**Comment by deltanedas** [comment]
i want to eventually use this for pro plates, could also work for artifacts

**Comment by 0x6273** [comment]
Calling this `ItemPlacerComponent` is maybe not ideal since it works with any entity and not just items.

**Comment by deltanedas** [comment]
i dont like the name overall but im not good with naming things
it integrates with PlaceableSurface tho which is for items

**Review by mirrorcult — CHANGES_REQUESTED** [review]
why not entirely merge `ItemPlacer` with `PlaceableSurface`?

**Comment by mirrorcult** [comment]
code looks fine besides that though

**Comment by deltanedas** [comment]
doesnt make sense to track placed items for inert things like tables that are still placeable but have no logic

**Comment by mirrorcult** [comment]
fair enough i suppose

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**